src/entities/amm/amm.ts

```diff
@@ -1482,6 +1483,36 @@ export class AMM {
     return tickToFixedRate(closestUsableTick).toNumber();
   }
 
+  // historocal rates
+
+  public async getHistoricalVariableRate(
+    subgraphUrl: string,
+    filters: {
+      granularity: Granularity;
+      timeframeMs: number;
+    },
+  ): Promise<any[]> {
+    const result = await getHistoricalRates(
+      subgraphUrl,
+      false,
+      filters,
+      undefined,
+      this.rateOracle.id,
+    );
+    return result;
+  }
+
+  public async getHistoricalFixedRate(
+    subgraphUrl: string,
```

Comment:
replace with `chainId` and use `getSubgraphURL(chainId, SubgraphURLEnum.voltzProtocol)` to get the subgraph URL